internal/xds/translator/listener.go

```diff
@@ -401,7 +404,57 @@ func buildXdsDownstreamTLSSocket(tlsConfigs []*ir.TLSListenerConfig) (*corev3.Tr
 	}, nil
 }
 
-func buildXdsDownstreamTLSSecret(tlsConfig *ir.TLSListenerConfig) *tlsv3.Secret {
+func buildTLSParams(tlsConfig *ir.TLSListenerConfig) *tlsv3.TlsParameters {
+	p := &tlsv3.TlsParameters{}
+	isEmpty := true
+	if tlsConfig.MinVersion != nil {
+		p.TlsMinimumProtocolVersion = buildTLSVersion(tlsConfig.MinVersion)
+		isEmpty = false
+	}
+	if tlsConfig.MaxVersion != nil {
+		p.TlsMaximumProtocolVersion = buildTLSVersion(tlsConfig.MaxVersion)
+		isEmpty = false
+	}
+	if len(tlsConfig.Ciphers) > 0 {
+		p.CipherSuites = tlsConfig.Ciphers
+		isEmpty = false
+	}
+	if len(tlsConfig.ECDHCurves) > 0 {
+		p.EcdhCurves = tlsConfig.ECDHCurves
+		isEmpty = false
+	}
+	if len(tlsConfig.SignatureAlgorithms) > 0 {
+		p.SignatureAlgorithms = tlsConfig.SignatureAlgorithms
+		isEmpty = false
+	}
+	if isEmpty {
+		return nil
+	}
+	return p
+}
+
+func buildTLSVersion(version *gwv1a1.TLSVersion) tlsv3.TlsParameters_TlsProtocol {
+	lookup := map[gwv1a1.TLSVersion]tlsv3.TlsParameters_TlsProtocol{
+		gwv1a1.TLSv10: tlsv3.TlsParameters_TLSv1_0,
+		gwv1a1.TLSv11: tlsv3.TlsParameters_TLSv1_1,
+		gwv1a1.TLSv12: tlsv3.TlsParameters_TLSv1_2,
+		gwv1a1.TLSv13: tlsv3.TlsParameters_TLSv1_3,
+	}
+	if r, found := lookup[*version]; found {
+		return r
+	}
+	return tlsv3.TlsParameters_TLS_AUTO
+}
+
+func buildALPNProtocols(in []gwv1a1.ALPNProtocol) []string {
```

Comment:
imo this is not needed if we use `[]string` in the IR

Reply:
Changed.